sympy/ntheory/generate.py

```diff
@@ -39,16 +44,36 @@ class Sieve:
     """
 
     # data shared (and updated) by all Sieve instances
-    _list = _array('l', [2, 3, 5, 7, 11, 13])
+    def __init__(self):
+        self._n = 6
+        self._list = _aset(2, 3, 5, 7, 11, 13) # primes
+        self._tlist = _aset(0, 1, 1, 2, 2, 4) # totient
+        self._mlist = _aset(0, 1, -1, -1, 0, -1) # mobius
+        assert all(len(i) == self._n for i in (self._list, self._tlist, self._mlist))
 
     def __repr__(self):
-        return "<Sieve with %i primes sieved: 2, 3, 5, ... %i, %i>" % \
-            (len(self._list), self._list[-2], self._list[-1])
-
-    def _reset(self):
+        return ("<Sieve with \n%s (%i): %i, %i, %i, ... %i, %i\n"
+             "%s (%i): %i, %i, %i, ... %i, %i\n"
+             "%s (%i): %i, %i, %i, ... %i, %i>") % (
+             'primes', len(self._list),
+                 self._list[0], self._list[1], self._list[2],
+                 self._list[-2], self._list[-1],
+             'totient', len(self._tlist),
+                 self._tlist[0], self._tlist[1],
+                 self._tlist[2], self._tlist[-2], self._tlist[-1],
+             'mobius', len(self._mlist),
+                 self._mlist[0], self._mlist[1],
+                 self._mlist[2], self._mlist[-2], self._mlist[-1])
+
+    def _reset(self, prime=True, totient=True, mobius=True):
```

Comment:
Or to make it easier to do only one,
```
def _reset(self, prime=None, totient=None, mobius=None):
    """reset all caches (default). To reset one or more set the
    desired keyword to True."""
    if all(i is None for i in 'prime totient mobius'.split()):
        prime = totient = mobius = True
...
```

Reply:
Did you mean `in (prime, totient, mobius)` (as `.split()` will give strings)?
And whats the advantage of this over the current method (with default value `True`)?